### PR TITLE
Add enumeration of block devices to ec2_block_devices

### PR DESCRIPTION
**Pull request by ghost**
##### Issue Type:
- Feature Pull Request
##### Ansible Version:

```
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Summary:

This change allows you to recover the ec2 block device mappings for a given instance.
##### Example output:

Produces output as below to enable you to make use of **vol IDs** from ec2.

```
        "ec2_block_devices": {
          "sda1": "vol-!!!!!!!!"
        }
```

### CODE REVIEW

**Comment by aioue** [comment]
+1

**Comment by ghost** [comment]
@bcoca any chance of getting the merged?

File: contrib/inventory/ec2.py [hunk 1]
```diff
@@ -1144,6 +1145,10 @@ def get_host_info_dict_from_instance(self, instance):
                     group_names.append(group.name)
                 instance_vars["ec2_security_group_ids"] = ','.join([str(i) for i in group_ids])
                 instance_vars["ec2_security_group_names"] = ','.join([str(i) for i in group_names])
+            elif key == 'ec2_block_device_mapping':
+                instance_vars["ec2_block_devices"] = {}
+                for k, v in value.iteritems():
```
Comment (abadger) [inline contrib/inventory/ec2.py]: For python3 compat, avoid iteritems unless you're dealing with large data sets.  If you are, then use ansible.compat.six.iteritems.

File: contrib/inventory/ec2.py [hunk 2]
```diff
@@ -1144,6 +1145,10 @@ def get_host_info_dict_from_instance(self, instance):
...
+                    instance_vars["ec2_block_devices"][ ( k.split( '/' ) )[ -1 ] ] = v.volume_id
```
Comment (abadger) [inline contrib/inventory/ec2.py]: Probably won't make any difference here but there's a stdlib function to split filename from directory name:

``` python
import os.path
os.path.basename(k)
```

or you can make this more targeted by doing it like this:

``` python
instance_vars["ec2_block_devices"][ k.rsplit( '/' )[ -1 ] ] = v.volume_id
```

**Comment by abadger** [comment]
Just looking at the code but not knowing ec2 at all, this looks good to merge.  I mentioned a few minor cleanups that can be easily changed post-merge if @khushil doesn't get to them first.  @bcoca, @ryansb do either of you want to put an approval stamp on this from the ec2 POV?

**Comment by ryansb** [comment]
Other than your comments, looks good from an EC2 viewpoint.

**Comment by MerlinDMC** [comment]
Rebased the changes on the current devel branch

**Pull request merged**